[skip ci] Update README.md - Add Featured Models list. (#25452)
tenstorrent/tt-metal: README.md

> **Last Update:** June 23, 2025
>
> **Notes:**
>
> - ttft = time to first token | t/s/u = tokens/second/user | t/s = tokens/second; where t/s = t/s/u * batch.
> - TP = Tensor Parallel, DP = Data Parallel; Defines parallelization factors across multiple devices.
> - The reported LLM performance is for an input sequence length (number of rows filled in the KV cache) of 128 for all models except Mamba (which can accept any sequence length).
> - The t/s/u reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.
> - Performance numbers were collected using the tt-metal model demos (accessible via the model links). If running with a vLLM inference server, performance may be different.
> - \* Blackhole software optimization is under active development.  Please join us in shaping the future of open source AI! <br> [\[Discord\]](https://discord.gg/tenstorrent) [\[Developer Hub\]](https://tenstorrent.com/developers)

> > - For more information regarding vLLM installation and environment creation visit the [Tenstorrent vLLM repository](https://github.com/tenstorrent/vllm/blob/dev/tt_metal/README.md).

Reviewer: These footnotes are still relevant and should be kept.

Author: Footnotes have been included.

> | 32    | [Galaxy (Wormhole)](https://tenstorrent.com/hardware/galaxy) | 105      | 59.1  | 80              | 1891.2  | [v0.59.0-rc52](https://github.com/tenstorrent/tt-metal/tree/v0.59.0-rc52) | [f028da1](https://github.com/tenstorrent/vllm/tree/f028da11b5b8205272bf18a478de93bd2dd3e29e/tt_metal) |

Reviewer: Some numbers were updated recently in https://github.com/tenstorrent/tt-metal/commit/a335bb0a2a9ddf4d14388a513a8ec894c021621f, could you apply the same changes here?

Author: Release link has been updated.

Reviewer: The entire row needs to be copied (perf numbers and commits)

Author: Row updated with new perf numbers/release info.

| [0.57.0](https://github.com/tenstorrent/tt-metal/releases/tag/v0.57.0) | Apr 15, 2025 |
| [0.56.0](https://github.com/tenstorrent/tt-metal/releases/tag/v0.56.0) | Mar 7, 2025 |

## Featured Models

The models team is focused on developing the following models into a customer ready state; optimizations to achieve performance, accuracy, and compatibility targets are ongoing, follow each model link for more information.

> Performance Metrics Notes:

Author: I added the LLM performance comment back in.

> Performance Metrics Notes:

Reviewer: Thanks, also the part about `(number of rows filled in the KV cache)` and Mamba can be removed

| [0.57.0](https://github.com/tenstorrent/tt-metal/releases/tag/v0.57.0) | Apr 15, 2025 |
| [0.56.0](https://github.com/tenstorrent/tt-metal/releases/tag/v0.56.0) | Mar 7, 2025 |

## Featured Models

The models team is focused on developing the following models into a customer ready state; optimizations to achieve performance, accuracy, and compatibility targets are ongoing, follow each model link for more information.

Performance Metrics Notes:

> - Time to first token (TTFT) descibes in milliseconds how long it takes for the model to generate the first token.

Reviewer 2: ```suggestion
- Time to First Token (TTFT): Measures the time (in milliseconds) it takes to generate the first output token after input is received.
```

Author: Changes included.

> - T/S/U measures tokens per second per user. The T/S/U reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.

Reviewer 2: ```suggestion
- T/S/U (Tokens per Second per User): Represents the throughput of first-token generation after prefill. It is calculated as 1 / inter-token latency.
```

> - T/S measures tokens per second, where T/S = T/S/U * batch.

Reviewer 2: ```suggestion
- T/S (Tokens per Second): Represents total token throughput, calculated as T/S = T/S/U × batch size.
```

Author: Changes included.

> - Tensor Parallel (TP) and Data Parallel (DP) define parallelization factors across mulitiple devices.

Reviewer 2: ```suggestion
- TP (Tensor Parallel) and DP (Data Parallel): Indicate the parallelization factors across multiple devices.
```

- Reported LLM performance is for an input sequence length of 128 for all models.

> - Performance numbers were collected using the tt-metal model demos (accessible via the model links). If running with a vLLM inference server, performance may be different.

Author: Changes included.
